README.md

```diff
@@ -41,6 +37,61 @@ grails-profile
 ---------
 _Todo_: Add the docs
 
+grails-profile-publish
+---------
+_Todo_: Add the docs
+
+grails-publish
+---------
+A Gradle plugin to simplify publishing:
+
+1. snapshots with either the Maven Publish gradle plugin or Nexus Publish gradle plugin.
+2. releases with the Nexus Publish gradle plugin.
+
+Example Usage:
+
+    grailsPublish {
+        websiteUrl = 'http://foo.com/myplugin'
+        license {
+            name = 'Apache-2.0'
+        }
+        issueTrackerUrl = 'https://github.com/myname/myplugin/issues'
+        vcsUrl = 'https://github.com/myname/myplugin'
+        title = 'My plugin title'
+        desc = 'My plugin description'
+        developers = [johndoe: 'John Doe']
+    }
+
+or
+
+    grailsPublish {
+        githubSlug = 'foo/bar'
+        license {
+            name = 'Apache-2.0'
+        }
+        title = 'My plugin title'
+        desc = 'My plugin description'
+        developers = [johndoe: 'John Doe']
+    }
+
+By default this plugin will publish to the specified `MAVEN_PUBLISH` instance for snapshots, and `NEXUS_PUBLISH` for releases.  To change the snapshot publish behavior, you can set the `snapshotRepoType` to `RepositoryTarget.NEXUS_PUBLISH`.
+
+The credentials and connection url must be specified as a project property or an environment variable.
+
+The `MAVEN_PUBLISH` Environment Variables are:
+
+    ARTIFACTORY_USERNAME
+    ARTIFACTORY_PASSWORD
+    ARTIFACTORY_URL
+
+The `NEXUS_PUBLISH` Environment Variables are:
+
+    SONATYPE_NEXUS_URL
+    SONATYPE_SNAPSHOT_URL
+    SONATYPE_USERNAME
+    SONATYPE_PASSWORD
+    SONATYPE_STAGING_PROFILE_ID
+
```

Comment:
I'm concerned about the clarity of our current repository terminology. We're using multiple terms that might confuse readers about their relationships:

Current Terms:
- `RepositoryTarget`
- `snapshotRepoType` (target vs type, they are all Maven (type) repos right? So we are targeting different repos)
- `MAVEN_PUBLISH`
- `NEXUS_PUBLISH`
- `ARTIFACTORY_*`
- `SONATYPE_*`

`MAVEN_PUBLISH` vs `NEXUS_PUBLISH` is counterintuitive (to me, if I was a new user), as I would use `NEXUS_PUBLISH` to publish to Maven Central.

Does it make sense with `RepositoryTarget.GENERIC` and `RepositoryTarget.NEXUS` with environment vars `PUBLISH_GENERIC_*` and `PUBLISH_NEXUS_*`?

Reply:
After giving this some more thought, I’d like to better understand the problem we’re solving here.
If I understand correctly, the goal is to publish snapshot and release artifacts to different repositories.

Would it make sense to name them `RepositoryTarget.SNAPSHOT` and `RepositoryTarget.RELEASE`?
Similarly, we could use environment variables like `PUBLISH_SNAPSHOT_*` and `PUBLISH_RELEASE_*`.

It’s getting late here, so there’s a chance I’m overlooking something obvious 😄. Let me know your thoughts!

Reply:
We could make signing an attribute, `default = true` for release and `false` for snapshot.
Depending on the repo url, use the nexus publish plugin or not.

Reply:
couple of problems:

1. every other plugin will likely only publish to maven central for snapshots & releases
2. grails currently publishes to artifactory (repo.grails.org) for snapshots (valid maven repo)
3. eventually we may want to use maven central's snapshot location for grails; in some case we already did this (rest plugin)
4. plugins may even want to publish to github, which only requires maven publish

The setup for maven central is cumbersome - especially in multiproject builds due to its signing requirements and the close/stage steps.  There are a lot of legacy plugins that would benefit from an easier to configure setup like grailsPublish - we had to make a blog post because it was so complicated.

My goal was:
1. to make this plugin public since the majority of the logic already is here.  That then allows people to easily migrate older plugins to support publishing with minimal configuration.  
2. support multiproject builds
3. cleanup the legacy fields that no longer apply (they were left around when it was internalized)
4. have a way for us to easily switch snapshot publish locations (just change the type and the appropriate url from the org would be used)
5. This also would consolidate all of our publishing logic to one location, this plugin.

As part of the milestone we could announce this and a sample github action to make migrating easier.

Reply:
To your snapshot/ release target comments, that does not makes sense because its valid to publish snapshots to a maven repo or maven central's snapshot location.  We ultimately are configuring both the snapshot and release locations - which determines which plugin we use.

Reply:
At the heart of this plugin we are making it use different gradle plugins.  Maybe the RepositoryTarget should be based on that

Reply:
@matrei can you check this again?  See if this is better.